### ISSUE TITLE
Rollup retains classes mentioned in fallthru after throw

### ISSUE THREAD

**Issue by lambdageek** [rollup/rollup#4542]
### Rollup Version

2.75.7

### Operating System (or Browser)

Chrome

### Link To Reproduction

https://rollupjs.org/repl/?version=2.75.7&shareable=JTdCJTIybW9kdWxlcyUyMiUzQSU1QiU3QiUyMm5hbWUlMjIlM0ElMjJtYWluLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMiUyRiolMjBUUkVFLVNIQUtJTkclMjAqJTJGJTVDbmltcG9ydCUyMCU3QiUyMG1ha2VDJTIwJTdEJTIwZnJvbSUyMCcuJTJGbWF0aHMuanMnJTNCJTVDbiU1Q25jb25zb2xlLmxvZyglMjBtYWtlQyglMjAlNUMlMjJhYmMlNUMlMjIlMjApLm15U3RyaW5nJTIwKSUzQiUyMCUyMiUyQyUyMmlzRW50cnklMjIlM0F0cnVlJTdEJTJDJTdCJTIybmFtZSUyMiUzQSUyMm1hdGhzLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMiUyRiUyRiUyMG1hdGhzLmpzJTVDbiU1Q25jb25zdCUyMGNvbmRpdGlvbiUyMCUzRCUyMGZhbHNlJTNCJTVDbiU1Q25jbGFzcyUyMEMlMjAlN0IlNUNuJTIwJTIwY29uc3RydWN0b3IocyklMjAlN0IlNUNuJTVDdCU1Q3R0aGlzLm15U3RyaW5nJTIwJTNEJTIwcyUzQiU1Q24lNUN0JTdEJTVDbiU3RCU1Q24lNUNuJTJGJTJGJTIwVGhpcyUyMGZ1bmN0aW9uJTIwZ2V0cyUyMGluY2x1ZGVkJTVDbmV4cG9ydCUyMGZ1bmN0aW9uJTIwbWFrZUMlMjAoJTIwcyUyMCklMjAlN0IlNUNuJTVDdGlmJTIwKCFjb25kaXRpb24pJTIwJTdCJTVDbiU1Q3QlNUN0dGhyb3clMjBuZXclMjBFcnJvciUyMCglNUMlMjJjb25kaXRpb24lMjBpcyUyMGZhbHNlJTVDJTIyKSUzQiU1Q24lNUN0JTdEJTVDbiU1Q3QlMkYlMkYlMjBlbHNlJTIwJTJGJTJGJTIwdW5jb21tZW50JTIwdGhpcyUyMCdlbHNlJyUyMGFuZCUyMEMlMjBpcyUyMG5vdCUyMHJldGFpbmVkJTVDbiU1Q3RyZXR1cm4lMjBuZXclMjBDKHMpJTNCJTVDbiU3RCUyMiUyQyUyMmlzRW50cnklMjIlM0FmYWxzZSU3RCU1RCUyQyUyMm9wdGlvbnMlMjIlM0ElN0IlMjJmb3JtYXQlMjIlM0ElMjJpaWZlJTIyJTJDJTIybmFtZSUyMiUzQSUyMm15QnVuZGxlJTIyJTJDJTIyYW1kJTIyJTNBJTdCJTIyaWQlMjIlM0ElMjIlMjIlN0QlMkMlMjJnbG9iYWxzJTIyJTNBJTdCJTdEJTdEJTJDJTIyZXhhbXBsZSUyMiUzQW51bGwlN0Q=

### Expected Behaviour

In a module like this:

```javascript

const condition = false;

class C {
	constructor(s) {
		this.myString = s;
	}
}

// This function gets included
export function makeC ( s ) {
	if (!condition) {
		throw new Error ("condition is false");
	}
	// else // uncomment this 'else' and C is not retained
	return new C(s);
}

```

I would expect that the result of tree shaking would not retain the definition of `C` because it is only used in the fallthru case after a `throw`.

Indeed that's how it works if the `new C` is in an `else` branch. the defintion of `C` is dropped.

### Actual Behaviour

the call to `new C` is gone, but the definition of `C` is still present

```javascript
	// maths.js

	class C {
		constructor(s) {
			this.myString = s;
		}
	}

	// This function gets included
	function makeC ( s ) {
		{
			throw new Error ("condition is false");
		}
	}
```

**Comment by lukastaegert** [comment]
Turns out we were unnecessarily including return expressions of all used functions even if the return was never reached. Fix at #4573

**Comment by lukastaegert** [comment]
This is your example via #4573: https://rollupjs.org/repl/?pr=4573&shareable=JTdCJTIybW9kdWxlcyUyMiUzQSU1QiU3QiUyMm5hbWUlMjIlM0ElMjJtYWluLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMiUyRiolMjBUUkVFLVNIQUtJTkclMjAqJTJGJTVDbmltcG9ydCUyMCU3QiUyMG1ha2VDJTIwJTdEJTIwZnJvbSUyMCcuJTJGbWF0aHMuanMnJTNCJTVDbiU1Q25jb25zb2xlLmxvZyglMjBtYWtlQyglMjAlNUMlMjJhYmMlNUMlMjIlMjApLm15U3RyaW5nJTIwKSUzQiUyMCUyMiUyQyUyMmlzRW50cnklMjIlM0F0cnVlJTdEJTJDJTdCJTIybmFtZSUyMiUzQSUyMm1hdGhzLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMiUyRiUyRiUyMG1hdGhzLmpzJTVDbiU1Q25jb25zdCUyMGNvbmRpdGlvbiUyMCUzRCUyMGZhbHNlJTNCJTVDbiU1Q25jbGFzcyUyMEMlMjAlN0IlNUNuJTIwJTIwY29uc3RydWN0b3IocyklMjAlN0IlNUNuJTVDdCU1Q3R0aGlzLm15U3RyaW5nJTIwJTNEJTIwcyUzQiU1Q24lNUN0JTdEJTVDbiU3RCU1Q24lNUNuJTJGJTJGJTIwVGhpcyUyMGZ1bmN0aW9uJTIwZ2V0cyUyMGluY2x1ZGVkJTVDbmV4cG9ydCUyMGZ1bmN0aW9uJTIwbWFrZUMlMjAoJTIwcyUyMCklMjAlN0IlNUNuJTVDdGlmJTIwKCFjb25kaXRpb24pJTIwJTdCJTVDbiU1Q3QlNUN0dGhyb3clMjBuZXclMjBFcnJvciUyMCglNUMlMjJjb25kaXRpb24lMjBpcyUyMGZhbHNlJTVDJTIyKSUzQiU1Q24lNUN0JTdEJTVDbiU1Q3QlMkYlMkYlMjBlbHNlJTIwJTJGJTJGJTIwdW5jb21tZW50JTIwdGhpcyUyMCdlbHNlJyUyMGFuZCUyMEMlMjBpcyUyMG5vdCUyMHJldGFpbmVkJTVDbiU1Q3RyZXR1cm4lMjBuZXclMjBDKHMpJTNCJTVDbiU3RCUyMiUyQyUyMmlzRW50cnklMjIlM0FmYWxzZSU3RCU1RCUyQyUyMm9wdGlvbnMlMjIlM0ElN0IlMjJmb3JtYXQlMjIlM0ElMjJlcyUyMiUyQyUyMm5hbWUlMjIlM0ElMjJteUJ1bmRsZSUyMiUyQyUyMmFtZCUyMiUzQSU3QiUyMmlkJTIyJTNBJTIyJTIyJTdEJTJDJTIyZ2xvYmFscyUyMiUzQSU3QiU3RCU3RCUyQyUyMmV4YW1wbGUlMjIlM0FudWxsJTdE
